### PR TITLE
[Calyx] Support lowering of `arith.select`

### PR DESCRIPTION
**Pull request by rachitnigam**
Implements lowering for `arith.select` using Calyx's `std_mux` primitive.

### CODE REVIEW

**Comment by cgyurgyik** [comment]
Can you add tests for lowering / emitter to verify this works as expected?

**Comment by rachitnigam** [comment]
Good call! Done. Also N00b question: what is the workflow for updating lit tests? For these, I manually (and very slowly) copied over the lines generated from the error. Is there a way to just run the command that `lit` is going to run and see the output for each test?

**Review by mikeurbach — APPROVED** [review]
This looks good to me.

> what is the workflow for updating lit tests? For these, I manually (and very slowly) copied over the lines generated from the error. Is there a way to just run the command that lit is going to run and see the output for each test?

I generally run the RUN command from the test manually with no changes, then add my new input IR to the test, then run it again, and find how my new outputs show up and copy them in as CHECK lines (using [substitutions](https://llvm.org/docs/CommandGuide/FileCheck.html#filecheck-string-substitution-blocks) as relevant, although we aren't doing that very well in the tests you updated here). So, it's a little manual, and there's no silver bullet. There is an MLIR [tool](https://github.com/llvm/llvm-project/blob/main/mlir/utils/generate-test-checks.py) to regenerate filecheck tests automatically, but it only really works if you always use that tool for the same test, and we aren't using it here.

File: include/circt/Dialect/Calyx/CalyxPrimitives.td [hunk 1]
```diff
@@ -267,6 +267,10 @@ def AndLibOp  : CombinationalArithBinaryLibraryOp<"and"> {}
 def OrLibOp   : CombinationalArithBinaryLibraryOp<"or"> {}
 def XorLibOp  : CombinationalArithBinaryLibraryOp<"xor"> {}
 
+def MuxLibOp : CalyxLibraryOp<"mux", [Combinational, SameTypeConstraint<"tru", "fal">, SameTypeConstraint<"tru", "out">]> {
```
Comment (mortbopet) [inline include/circt/Dialect/Calyx/CalyxPrimitives.td:270]: nit: line length, i would usually split the trait list to the next line.

**Review by cgyurgyik — APPROVED** [review]

File: include/circt/Dialect/Calyx/CalyxPrimitives.td (hunk 1 above)
Comment (rachitnigam) [reply include/circt/Dialect/Calyx/CalyxPrimitives.td:270]: Done!

**Pull request merged**